### PR TITLE
Netty client support for authorized proxy

### PR DESCRIPTION
**Pull request by guillepb10**
## Description
This change able the use of a secured corporate proxy to communicate with aws, into the netty async client.

## Motivation and Context
So many corporations used a proxy to connect to internet. Besides this corporate usually needs a basic autentication to connect to the proxy. Until nowadays, it was impossible to used a secured coroporate proxy with netty client, couse it did not support to specify the user and password required to connect to proxy. This change makes it possible.

## Testing
It has been compleated unit testing of the library to ensure the purpose of the change it is successfully met.
Besides, I have tested the change with a personal project, where I try to connect to CloudWatchLogs api behind a coporate proxy, and it succeeded.

As cloud bee shown this is a very little change, that only affect to netty-nio-client. So it must not break the current work of the client.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by guillepb10** [comment]
Have I any way of check why AWS CoudeBuilds checks are failing?

**Comment by guillepb10** [comment]
@dagnir First of all, sorry for disturb, but I have opened this pull request a week ago and I didn't have any feedback about it. Could you review it? Or give me any gap of time to be reviewed?

File: http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/ProxyTunnelInitHandlerTest.java [hunk 1]
```diff
@@ -52,6 +35,15 @@
...
 import org.mockito.runners.MockitoJUnitRunner;
 
+import java.io.IOException;
+import java.net.URI;
+import java.util.Base64;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
```
Comment (debora-ito) [inline http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/ProxyTunnelInitHandlerTest.java:45]: We don't use wildcard imports in general. We do provide a `intellij-codestyle.xml` that helps formatting the code automatically according to our code style guidelines in Intellij. 

Can you apply the codestyle file to your code? The file can be found here, along with other helpful information: https://github.com/aws/aws-sdk-java-v2/blob/master/docs/GettingStarted.md

**Review by debora-ito — COMMENTED** [review]
Made some comments in an initial review.

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/ProxyConfiguration.java [hunk 2]
```diff
@@ -94,6 +116,14 @@ public boolean equals(Object o) {
             return false;
         }
 
+        if (username != null ? !username.equals(that.host) : that.username != null) {
```
Comment (debora-ito) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/ProxyConfiguration.java:119]: Fix: `!username.equals(that.host)`

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/ProxyConfiguration.java [hunk 3]
```diff
@@ -94,6 +116,14 @@ public boolean equals(Object o) {
...
+            return false;
+        }
+
+        if (password != null ? !password.equals(that.host) : that.password != null) {
```
Comment (debora-ito) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/ProxyConfiguration.java:123]: Fix: `!password.equals(that.host)`

File: http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/ProxyTunnelInitHandlerTest.java (hunk 1 above)
Comment (debora-ito) [inline http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/ProxyTunnelInitHandlerTest.java:45]: Wilcard imports still here.

**Comment by guillepb10** [comment]
Hi @debora-ito , I think I have already solved your comments

**Comment by cenedhryn** [comment]
@guillepb10 Thanks for submitting the PR, and sorry it took a while for us to fully review it. I only had some minor comments.

**Comment by guillepb10** [comment]
Thank you for your feedback @cenedhryn. I've just updated the PR qith your comments.

**Review by cenedhryn — APPROVED** [review]

**Comment by debora-ito** [comment]
@all-contributors please add @guillepb10 for code.

**Pull request merged**